sdks/go/pkg/beam/runners/universal/runnerlib/job.go

```diff
@@ -0,0 +1,140 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runnerlib
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/apache/beam/sdks/go/pkg/beam"
+	"github.com/apache/beam/sdks/go/pkg/beam/core/runtime"
+	"github.com/apache/beam/sdks/go/pkg/beam/log"
+	jobpb "github.com/apache/beam/sdks/go/pkg/beam/model/jobmanagement_v1"
+	pb "github.com/apache/beam/sdks/go/pkg/beam/model/pipeline_v1"
+	"github.com/apache/beam/sdks/go/pkg/beam/provision"
+	"github.com/golang/protobuf/proto"
+)
+
+// JobOptions capture the various options for submitting jobs
+// to universal runners.
+type JobOptions struct {
+	// Name is the job name.
+	Name string
+	// Experiments are additional experiments.
+	Experiments []string
+
+	// TODO(herohde) 3/17/2018: add further parametrization as needed
+
+	// InternalJavaRunner is the class of the receiving Java runner. To be removed.
+	InternalJavaRunner string
+}
+
+// Prepare prepares a job to the given endpoint. It returns an id and endpoint, if successful.
```

Comment:
We may be able to improve this comment, as I'm having difficulty reconciling what's here, and in the implementation.

 Knowing nothing about what PrepareJob request on the service does, but reading this implementation, it looks like the idea is to prepare the runner to receive the job, largely by getting a place to stage artifacts (like the worker) to run the job later.
So, it's preparing the service for the job. Is that right?
With that in mind, and looking at how the returns are used in this PR perhaps:

// Prepare prepares a given Beam model runner to receive a job. If successful, Prepare returns an id, and a URL
// endpoint where the runner expects job artifacts to be staged.

Reply:
Yes. Rephrased the description.